src/stmt.rs

```diff
@@ -116,7 +116,7 @@ fn format_stmt(
 
     let result = match stmt.kind {
         ast::StmtKind::Local(ref local) => local.rewrite(context, shape),
-        ast::StmtKind::Expr(ref ex) | ast::StmtKind::Semi(ref ex) => {
+        ast::StmtKind::Semi(ref ex) => {
```

Comment:
was this the cause of the double semicolon counting?

Reply:
See the comment below; I pointed out the reason for the the double semicolon counting in more detail.

I split these two branches because we need to preserve semicolons only if we have a `StmtKind::Semi`, and not if we have a `StmtKind::Expr`.